projects/storefrontlib/src/shared/components/anonymous-consents/dialog/anonymous-consents-dialog.component.ts

```diff
@@ -47,18 +48,60 @@ export class AnonymousConsentsDialogComponent implements OnInit, OnDestroy {
 
   rejectAll(): void {
     this.subscriptions.add(
-      this.anonymousConsentsService.withdrawAllConsents().subscribe()
+      combineLatest([this.templates$, this.consents$])
+        .pipe(
+          take(1),
+          distinctUntilChanged(),
+          tap(([templates, consents]) =>
+            templates.forEach(template => {
+              const consent = this.getCorrespondingConsent(template, consents);
+              if (this.anonymousConsentsService.isConsentGiven(consent)) {
```

Comment:
if state is null, and when no consents were given, it will always be false, therefore clicking rejectall/allow all in the dialog will never work